extensions/panache/panache-common/deployment/src/main/java/io/quarkus/panache/common/deployment/ByteCodeType.java

```diff
@@ -1,83 +1,55 @@
 package io.quarkus.panache.common.deployment;
 
-import static org.jboss.jandex.DotName.createSimple;
-import static org.objectweb.asm.Type.BOOLEAN_TYPE;
-import static org.objectweb.asm.Type.BYTE_TYPE;
-import static org.objectweb.asm.Type.CHAR_TYPE;
-import static org.objectweb.asm.Type.DOUBLE_TYPE;
-import static org.objectweb.asm.Type.FLOAT_TYPE;
-import static org.objectweb.asm.Type.INT_TYPE;
-import static org.objectweb.asm.Type.LONG_TYPE;
-import static org.objectweb.asm.Type.SHORT_TYPE;
-import static org.objectweb.asm.Type.getType;
-
+import java.util.HashMap;
+import java.util.Map;
 import java.util.StringJoiner;
 
+import org.jboss.jandex.ClassType;
 import org.jboss.jandex.DotName;
 import org.jboss.jandex.PrimitiveType;
-import org.objectweb.asm.Type;
-
-import io.quarkus.deployment.util.AsmUtil;
 
 public class ByteCodeType {
-    private final Type type;
-
-    public ByteCodeType(Type type) {
-        this.type = type;
+    private static final Map<String, org.jboss.jandex.Type> WRAPPER_TO_PRIMITIVE = new HashMap<>();
+
+    static {
+        WRAPPER_TO_PRIMITIVE.put(Boolean.class.getName(), PrimitiveType.BOOLEAN);
+        WRAPPER_TO_PRIMITIVE.put(Byte.class.getName(), PrimitiveType.BYTE);
+        WRAPPER_TO_PRIMITIVE.put(Short.class.getName(), PrimitiveType.SHORT);
+        WRAPPER_TO_PRIMITIVE.put(Integer.class.getName(), PrimitiveType.INT);
+        WRAPPER_TO_PRIMITIVE.put(Long.class.getName(), PrimitiveType.LONG);
+        WRAPPER_TO_PRIMITIVE.put(Float.class.getName(), PrimitiveType.FLOAT);
+        WRAPPER_TO_PRIMITIVE.put(Double.class.getName(), PrimitiveType.DOUBLE);
+        WRAPPER_TO_PRIMITIVE.put(Character.class.getName(), PrimitiveType.CHAR);
     }
```

Comment:
Minor suggestion to avoid unnecessary buckets in the HashMap
```suggestion
    private static final Map<String, org.jboss.jandex.Type> WRAPPER_TO_PRIMITIVE = Map.of(
            Boolean.class.getName(), PrimitiveType.BOOLEAN,
            Byte.class.getName(), PrimitiveType.BYTE,
            Short.class.getName(), PrimitiveType.SHORT,
            Integer.class.getName(), PrimitiveType.INT,
            Long.class.getName(), PrimitiveType.LONG,
            Float.class.getName(), PrimitiveType.FLOAT,
            Double.class.getName(), PrimitiveType.DOUBLE,
            Character.class.getName(), PrimitiveType.CHAR
    );
```

Reply:
Good idea, thanks!

Reply:
Done.

Reply:
Technically speaking, the implementation behind `Map.of()` is not particularly memory efficient, it maintains a load factor of 0.5. Using a `HashMap` would likely be smaller (default load factor is 0.75). In this particular case, the number of key/value pairs we store is 8, the number of elements we store into the array is 16, and the underlying array will have a length of 32. The map here is constant, so it would be possible to measure the collision rate and collision chain sizes, but I'm too lazy to do that. The reason why I think `Map.of()` is better isn't memory size -- it's just that the code is easier on the eyes :-)

Reply:
sure, it's easier on the eyes. But I wasn't referring to memory efficiency: the implementation requires a modulo operation rather than using a shift.
I don't know when this map is used and how much so it might not be worth to even talk about this, just pointing it out...

Reply:
Ah, you mean the process of clamping the computed hash code into the range `[0, table.length)`? Where this immutable map uses `hash % table.length`, while `HashMap` uses `hash & table.length - 1` because `table.length` there is always a power of 2? Yeah, that is a tiny bit more expensive, but this map is only used in the `unbox()` method, which has exactly 1 caller: `KotlinPanacheClassOperationGenerationVisitor.discoverTypeParameters()`. So I think we should be fine.